crates/build/src/lib.rs

```diff
@@ -38,13 +38,13 @@ async fn build_component(raw: RawComponentManifest, src: impl AsRef<Path>) -> Re
                 "Executing the build command for component {}: {}",
                 raw.id, b.command
             );
+            let workdir = construct_workdir(src.as_ref(), b.workdir.as_ref())?;
+            if !src.as_ref().starts_with(workdir.as_path()) {
```

Comment:
This won't catch `--workdir=..`. Perhaps it should be:
```suggestion
            if b.workdir.is_some() {
```
or just print the workdir unconditionally.

Reply:
The `workdir` returned by `construct_workdir` is absolutized, so it shouldn't contain dots. But that's implicit and your suggestion definitely makes the intent clearer.

Reply:
Right, and `workdir` is also absolutized at this point, so `--workdir=..` would always resolve to a prefix of `src`.